fedimint-api/src/db/mod.rs

```diff
@@ -684,6 +687,124 @@ where
     }
 }
 
+#[derive(Debug, Encodable, Decodable, Serialize)]
+pub struct DatabaseVersionKey;
+
+#[derive(Debug, Encodable, Decodable, Serialize, Clone)]
+pub struct DatabaseVersion {
+    pub version: u64,
+}
+
+impl DatabaseVersion {
+    pub fn new(version: u64) -> DatabaseVersion {
+        DatabaseVersion { version }
+    }
+}
+
+impl std::fmt::Display for DatabaseVersion {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{}", self.version)
+    }
+}
+
+impl DatabaseKeyPrefixConst for DatabaseVersionKey {
+    const DB_PREFIX: u8 = DbKeyPrefix::DatabaseVersion as u8;
+    type Key = Self;
+    type Value = DatabaseVersion;
+}
+
+impl std::fmt::Display for DbKeyPrefix {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+#[derive(Debug, Clone, Copy, Encodable, Decodable)]
+pub struct DatabaseVersionKeyPrefix;
+
+impl DatabaseKeyPrefixConst for DatabaseVersionKeyPrefix {
+    const DB_PREFIX: u8 = DbKeyPrefix::DatabaseVersion as u8;
+    type Key = DatabaseVersionKey;
+    type Value = DatabaseVersion;
+}
+
+#[repr(u8)]
+#[derive(Clone, EnumIter, Debug)]
+pub enum DbKeyPrefix {
+    DatabaseVersion = 0x50,
+    MigratableKeyValue = 0x51,
+}
+
+#[derive(Debug, Encodable, Decodable, DatabaseVersion)]
+#[struct_version(3, "key_1", "key_2")]
+#[struct_version(2, "key_1", "key_2")]
+#[struct_version(1, "key_1")]
+pub struct MigratableKey {
+    pub key_1: u64,
+
+    pub key_2: u64,
+
+    pub key_3: u64,
+}
+
+#[derive(Debug, Encodable, Decodable)]
+struct MigratableKeyPrefix;
+
+#[derive(Debug, Encodable, Decodable, Eq, PartialEq, DatabaseVersion)]
+#[struct_version(3, "value_1", "value_2", "value_3")]
+#[struct_version(2, "value_1", "value_2")]
+#[struct_version(1, "value_1")]
+pub struct MigratableValue {
```

Comment:
I kind of just expected a `MigratableValuev1`, `MigratableValuev2`, etc.

This way new code can just copy&paste old structs, add `vX` suffix and never have to touch them.

Reply:
See my other comment. If we think its overkill I can remove it and just rely on developers to create new versioned structs.

Reply:
Removed the custom derive macro